### DISCUSSION TITLE
Should Fedora alias dnf commands using /etc/alternatives instead of soft links?

### DISCUSSION THREAD

**Discussion by plvlnisse** [rpm-software-management/dnf5#1397] [Ideas]
Not at all sure if this is the right body to address this as it likely is a distribution issue rather than specific to this package, but in Redhat Fedora dnf commands are aliased to different releases using soft links in /usr/bin. I believe the distribution intent is to handle such things via /etc/alternatives? 

I have no real preference either way, but I prefer to be as consistent as possible within any particular distribution on how to do aliasing to simplify for the end user.

Below is from an old (2011) system continuously upgraded and most recently to Fedora 40 beta. dnf5 installed manually from Fedora40 repos after the version upgrade (it did not install automatically, but I guess the switch to dnf5 has been moved to release 41).

```
[slickers2(nsmeds):~] for f in /usr/bin/dnf*; do echo -e $(ls -ld $f)"\t::=> " $(rpm -qf $f);  done
lrwxrwxrwx 1 root root 5 Mar 29 01:00 /usr/bin/dnf -> dnf-3	::=>  dnf-4.19.2-1.fc40.noarch
-rwxr-xr-x 1 root root 2092 Mar 29 01:00 /usr/bin/dnf-3		::=>  python3-dnf-4.19.2-1.fc40.noarch
lrwxrwxrwx 1 root root 5 Mar 29 01:00 /usr/bin/dnf4 -> dnf-3	::=>  python3-dnf-4.19.2-1.fc40.noarch
-rwxr-xr-x 1 root root 1650968 Apr 3 02:00 /usr/bin/dnf5	::=>  dnf5-5.1.17-1.fc40.x86_64
-rwxr-xr-x 1 root root 1429 Mar 29 01:00 /usr/bin/dnf-automatic	::=>  dnf-automatic-4.19.2-1.fc40.noarch
```

**Comment by ppisar** [comment]
DNF5 developers discussed a form of the switch at lengths last year.
DNF5 developers has tested the symlink form last year in <https://copr.fedorainfracloud.org/coprs/rpmsoftwaremanagement/dnf5-testing/>.
Fedora approved using symlinks <https://fedoraproject.org/wiki/Changes/ReplaceDnfWithDnf5#Detailed_Description>.
I think the boat has sailed and Fedora will use symlinks.

**Comment by plvlnisse** [comment]
Thanks for reaching out and explaining. So this has been discussed and a conclusion has been made. 
I have no real preference so I am happy regardless.

I could not find the answer by searching (or I am just a bad searcher) so I appreciate your effort in explaining the background.
